### PR TITLE
GEODE-5727: rework how ResultModel deal with file contents.

### PR DESCRIPTION
**Pull request by jinmeiliao**
* file saving should be handled by the command's postExecutor to save to appropriate places
* after saving file content to a directory already, the ResultModel turned the FileResultModel into InfoResultModel.
* gfsh should not be the place to save files in ModelCommandResult.

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-web/src/integrationTest/java/org/apache/geode/management/internal/cli/commands/CommandOverHttpTest.java [hunk 1]
```diff
@@ -86,7 +86,7 @@ public void deployJar() throws Exception {
   public void exportConfig() throws Exception {
     String dir = temporaryFolder.getRoot().getAbsolutePath();
     gfshRule.executeAndAssertThat("export config --dir=" + dir).statusIsSuccess()
-        .containsOutput("Downloading Cache XML file: server-cache.xml")
-        .containsOutput("Downloading properties file: server-gf.properties");
+        .containsOutput("File saved to " + dir + "/server-cache.xml")
+        .containsOutput("File saved to " + dir + "/server-gf.properties");
```
Comment (pdxrunner) [inline geode-web/src/integrationTest/java/org/apache/geode/management/internal/cli/commands/CommandOverHttpTest.java]: Will this test pass on Windows?

File: geode-core/src/distributedTest/java/org/apache/geode/management/internal/cli/commands/ExportClusterConfigurationCommandDUnitTest.java [hunk 2]
```diff
@@ -86,4 +94,60 @@ public void getClusterConfigWithFile() throws IOException {
...
+    Path exportedZipPath =
+        tempFolder.getRoot().toPath().resolve("exportedCC.zip").toAbsolutePath();
+
+    testExportClusterConfig(exportedZipPath.toString());
+  }
+
+  @Test
+  public void testExportWithRelativePath() throws Exception {
+    testExportClusterConfig("mytemp/exportedCC.zip");
+    FileUtils.deleteQuietly(new File("mytemp"));
```
Comment (pdxrunner) [inline geode-core/src/distributedTest/java/org/apache/geode/management/internal/cli/commands/ExportClusterConfigurationCommandDUnitTest.java]: For this and the following test `mytemp` is created in the `tempFolder` so explicit deletion shouldn't be necessary.

Comment (jinmeiliao) [reply geode-core/src/distributedTest/java/org/apache/geode/management/internal/cli/commands/ExportClusterConfigurationCommandDUnitTest.java]: actually the exported files without absolute paths is written to the user.dir, so we need an explicit delete both here and the test below. The test below is deleting the wrong file. I checked a fix for that.

Comment (pdxrunner) [reply geode-core/src/distributedTest/java/org/apache/geode/management/internal/cli/commands/ExportClusterConfigurationCommandDUnitTest.java]: Thanks. I stand corrected.

**Pull request merged**